dvc/remote/webdav.py

```diff
@@ -0,0 +1,54 @@
+from .http import RemoteHTTP
+from dvc.scheme import Schemes
+
+import os.path
+
+from dvc.progress import Tqdm
+from dvc.exceptions import HTTPError
+
+
+class RemoteWEBDAV(RemoteHTTP):
+    scheme = Schemes.WEBDAV
+
+    def __init__(self, repo, config):
+        super().__init__(repo, config)
+
+        url = config.get("url")
+        if url:
+            self.path_info = self.path_cls(url)
+            self.path_info.scheme = self.path_info.scheme.replace(
```

Comment:
So far looks like this simply copies HTTP implementation. Are there any differences?

Reply:
Here I added only to change scheme. I was not sure if there would be problems in the future if I did it after. 
May be:
```
    def __init__(self, repo, config):
        super().__init__(repo, config)

        if self.path_info is not None:
            self.path_info.scheme = self.path_info.scheme.replace(
                "webdav", "http")
```

Reply:
@shizacat How does a typical webdav URL look like? Does it use `webdav` as a schema? If there are no differences, maybe let's just document that if one wants to use webdav he could access it as a HTTP(s) remote?

Reply:
He does not have his own separate scheme. This is a normal HTTP request.
It differs from the remote.http implementation in the following:
- To upload, you need another method (PUT, instead of POST). As a solution - add an option when configuring remote.http
- The creation of interstitial directories on the server may be required. Depends on the server settings, may not be critical if specified as a requirement. The same nginx allows this.

Reply:
Is there any particular way of detecting if PUT or POST is needed? Maybe some specific error is raised when you use one or another and you could fallback to another one?

I'm not familiar with webdav, so I'm just wondering if we should make it work as a generic HTTP(s) or if we should create a special remote for it. So far not 100% clear to me.

Reply:
There is no 100% way to determine the webdav server. There is a request through the OPTIONS method, but is not specified in the standard. I think it's better to separate them.
In recent commits, I made a separate _WebdavURLInfo_ object, and it already returns the correct url, there is no need to edit _base.py_.

Reply:
@shizacat Got it, thanks for clarifying! :pray: Looks better now. Btw, what about tests? Are there any options?